tests/bluetooth/controller/ctrl_cte_req/src/main.c

```diff
@@ -616,6 +616,743 @@ void test_cte_req_reject_inv_ll_param_peripheral_remote(void)
+	} else {
+		run_phy_update_peripheral(true, NULL, &phy_req, pu_event_counter(&conn),
+					  CONFIG_BT_CTLR_LLCP_PROC_CTX_BUF_NUM);
+	}
+}
+
+void test_central_local_phy_update_wait_for_cte_req_complete(void)
+{
+	test_local_phy_update_wait_for_cte_req_complete(BT_HCI_ROLE_CENTRAL);
+}
+
+void test_peripheral_local_phy_update_wait_for_cte_req_complete(void)
+{
+	test_local_phy_update_wait_for_cte_req_complete(BT_HCI_ROLE_PERIPHERAL);
+}
+
+static void run_remote_cte_req(struct pdu_data_llctrl_cte_req *cte_req)
+{
+	struct pdu_data_llctrl_cte_rsp remote_cte_rsp = {};
+	struct node_tx *tx;
+
+	/* The CTE response should already be enabled and request PDU should already be
+	 * received.
+	 */
+
+	/* Prepare */
+	event_prepare(&conn);
+
+	/* Tx Queue should have one LL Control PDU */
+	lt_rx(LL_CTE_RSP, &conn, &tx, &remote_cte_rsp);
+	lt_rx_q_is_empty(&conn);
+
+	/* TX Ack */
+	event_tx_ack(&conn, tx);
+
+	/* Done */
+	event_done(&conn);
+
+	/* Release tx node */
+	ull_cp_release_tx(&conn, tx);
+
+	/* There should not be a host notifications */
+	ut_rx_q_is_empty();
+}
+
+void test_phy_update_wait_for_remote_cte_req_complete(uint8_t role)
+{
+	struct pdu_data_llctrl_cte_req local_cte_req = {
+		.cte_type_req = BT_HCI_LE_AOA_CTE,
+		.min_cte_len_req = BT_HCI_LE_CTE_LEN_MIN,
+	};
+	struct pdu_data_llctrl_phy_req phy_req = { .rx_phys = PHY_CODED, .tx_phys = PHY_CODED };
+	uint8_t err;
+
+	phy_update_setup();
+
+	/* Role */
+	test_set_role(&conn, role);
+
+	/* Connect */
+	ull_cp_state_set(&conn, ULL_CP_CONNECTED);
+
+	/* Enable response for CTE request */
+	ull_cp_cte_rsp_enable(&conn, true, BT_HCI_LE_CTE_LEN_MAX,
+			      (BT_HCI_LE_AOA_CTE | BT_HCI_LE_AOD_CTE_1US | BT_HCI_LE_AOD_CTE_2US));
+
+	/* Prepare */
+	event_prepare(&conn);
+
+	/* Tx */
+	lt_tx(LL_CTE_REQ, &conn, &local_cte_req);
+
+	/* Done */
+	event_done(&conn);
+
+	/* Initiate an PHY Update Procedure */
+	err = ull_cp_phy_update(&conn, PHY_CODED, PREFER_S2_CODING, PHY_CODED, HOST_INITIATED);
+	zassert_equal(err, BT_HCI_ERR_SUCCESS, NULL);
+
+	run_remote_cte_req(&local_cte_req);
+
+	/* There should not be a host notifications */
+	ut_rx_q_is_empty();
+
+	zassert_equal(ctx_buffers_free(), CONFIG_BT_CTLR_LLCP_PROC_CTX_BUF_NUM - 1,
+		      "Free CTX buffers %d", ctx_buffers_free());
+
+	if (role == BT_HCI_ROLE_CENTRAL) {
+		run_phy_update_central(true, NULL, &phy_req, pu_event_counter(&conn),
+				       CONFIG_BT_CTLR_LLCP_PROC_CTX_BUF_NUM);
+	} else {
+		run_phy_update_peripheral(true, NULL, &phy_req, pu_event_counter(&conn),
+					  CONFIG_BT_CTLR_LLCP_PROC_CTX_BUF_NUM);
+	}
+}
+
+void test_central_phy_update_wait_for_remote_cte_req_complete(void)
+{
+	test_phy_update_wait_for_remote_cte_req_complete(BT_HCI_ROLE_CENTRAL);
+}
+
+void test_peripheral_phy_update_wait_for_remote_cte_req_complete(void)
+{
+	test_phy_update_wait_for_remote_cte_req_complete(BT_HCI_ROLE_PERIPHERAL);
+}
+
+void test_cte_req_wait_for_remote_phy_update_complete_and_disable(uint8_t role)
```

Comment:
Could these common tests that take a `role` parameter be made `static`?

Reply:
Yes, changed to static.